### PR TITLE
Markdown: Don't break a line at the place around Chinese or Japanese characters

### LINKED ISSUE #16710: Markdown: Disable breaking a line at the place around Chinese or Japanese characters even when `proseWrap` is `always`
**Prettier 3.3.3**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEBRKBzANgSwM4AWABIKemgNOaB5UeVZdXbQzU-c4y9UetvsXgvFDBwQAGhAQADjBzQ8yUAEMATkogB3AArKEclAqxqFATzliARkoVgA1nBgBlBQFs4AGRxRhSAGb6+5yxs7ewkrDwxkGCUAVzgxPiccSJi4kDgADwk4JRwXWH0AFWyoZRw4XV8sfxA8cKw4AEVoiHhkSuqAKzx0+zrG5tafP1SARwG4DVUJXRAFPABaTzgAExXRECiFHFxMAGEIJycFZDS8AFZ12sx6gEEYKJwzaPgNbPdPNuGxAhgnLAB1Ag4eB4UJCew6YE4ABuwKMJzAeFMIGhsQAklBVrB7GAclIbpj7DAjPVPlVUhJVHx-pYJCd9IYTOsPHwlDBJgoMEcydVQkpWScjkprMt1FB1pSPDB-jhljACMgACwABjESjgYxw6o5XOOQ3JYhgCjMMrlCqQACYxNE+AVjRUvmknGYVqtlq4FJhopy4AAxCBKI73cL054QEAAXwjQA)

```markdown
# Options (if any):
--prose-wrap=always --print-width=40
```

**Input:**

```md
English 日本語日本語日本語日本語日本語日本語日本語日本語日本語日本語日本語日本語日本語日本語日本語日本語日本語日本語 English sentence
```

**Output:**

```md
English 日本語日本語日本語日本語日本語日
本語日本語日本語日本語日本語日本語日本語
日本語日本語日本語日本語日本語日本語
English sentence

```

**Expected output:**

```md
English 日本語日本語日本語日本語日本語日本語日本語日本語日本語日本語日本語日本語日本語日本語日本語日本語日本語日本語 English
sentence
```

**Why?**

The result expected in #14936 breaks the existing Markdown documents. We should temporally disable breaking a line not only between Chinese/Japanese characters but also between Latin and Chinese/Japanese characters.
This issue is opened for closed by #16691.

### PR DESCRIPTION
**Pull request by tats-u**
Closes #16710

## Description

The 2nd safest part of https://github.com/prettier/prettier/pull/15081
All HTML browsers (even Firefox) insert space between Chinese/Japanese and others.

https://wpt.fyi/results/css/css-text/line-breaking?label=master&label=experimental&aligned&q=segment-break-transformation- (See only those that all browsers pass)

FYI, FIrefox treats the following characters in BMP and SSP (Plane 0-1) as Chinese/Japanese:

https://github.com/tats-u/cj-newline-test/?tab=readme-ov-file#result-in-firefox

The current behavior of Prettier (3.x) with `proseWrap` `always` conflicts with it.

This change will help Markdown documents converted to HTML or JS components to be rendered more correctly by browsers.

Note: this doesn't affect on line breaks between Chinese/Japanese and Korean letters because a space and a line break there is equivalent.

## Checklist

- [x] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory).
- [x] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

**Comment by tats-u** [comment]
This force push is just a rebase to `main`.
I will modify `print-width.js` to stop Prettier from wrap lines even between Chinese/Japanese characters.

File: tests/format/markdown/splitCjkText/__snapshots__/format.test.js.snap [hunk 1]
```diff
@@ -22,25 +22,20 @@ nasalカ゚キ゚ク゚ケ゚コ゚カ゚キ゚ク゚ケ゚コ゚カ゚キ゚ク
...
 
-nasalカ゚キ゚ク゚ケ゚コ゚カ゚キ゚ク゚ケ゚コ゚カ゚キ゚ク゚ケ゚コ゚カ゚キ゚ク゚
-ケ゚コ゚カ゚キ゚ク゚ケ゚コ゚カ゚キ゚ク゚ケ゚コ゚カ゚キ゚ク゚ケ゚コ゚カ゚キ゚ク゚
-ケ゚コ゚カ゚キ゚ク゚ケ゚コ゚カ゚キ゚ク゚ケ゚コ゚カ゚キ゚ク゚ケ゚コ゚
+nasalカ゚キ゚ク゚ケ゚コ゚カ゚キ゚ク゚ケ゚コ゚カ゚キ゚ク゚ケ゚コ゚カ゚キ゚ク゚ケ゚コ゚カ゚キ゚ク゚ケ゚コ゚カ゚キ゚ク゚ケ゚コ゚カ゚キ゚ク゚ケ゚コ゚カ゚キ゚ク゚ケ゚コ゚カ゚キ゚ク゚ケ゚コ゚カ゚キ゚ク゚ケ゚コ゚カ゚キ゚ク゚ケ゚コ゚
 
-かつてはワ行のワ、ヰ、ヱ、ヲに濁点を付して [v] 音を表現すること（ワ゛、ヰ゛、ヱ
-゛、ヲ゛）も行われたが、一般的にはならなかった。
+かつてはワ行のワ、ヰ、ヱ、ヲに濁点を付して [v]
+音を表現すること（ワ゛、ヰ゛、ヱ゛、ヲ゛）も行われたが、一般的にはならなかった。
```
Comment (tats-u) [inline tests/format/markdown/splitCjkText/__snapshots__/format.test.js.snap:38]: When `prev` or `next` is undefined, `" "` and `"\n"` have been interchangeable.

https://prettier.io/playground/#N4Igxg9gdgLgprEAuEhT00DTmg8qIKJQOYA2AlgM4AWGO+x5AOlJfboaWfQFTuWdPWuNROzGmU4CABsNbj6lKXSjjKMqPLJKskvuTnb1ykABoQEAA4wi0EslABDAE72IAdwAKDhNZS2Cz2wE9rYwAje1swAGs4GABlWwBbOAAZIig4ZAAzHxI4ELDI6JjTcNS8ZBh7AFdckBz4onKqmrgAD1M4eyJE2B8AFQ6oByI4LyyCHOMSUoI4AEVKiHhM7JqAKxIWmOm5haWkMYmQAEdduFcnUy8QWxIAWjS4ABMnoxAK2yJifABhCHj42zIEAjACsrym1DgAEEYBUiMFKvBXB0UmlluMamQYPECAB1MhEeAkYpgOAxTyEogAN0J-iBYBIQRAVOqAEkoM9YDEwJ1zFCOTEYP4ZujDqYnDlcWFTECfH5Aq9Ujl7DBzrY8ADRTVivZlUCAfYIo8XFBXuLUjBcURHjAyMgACwABmM9jgJyIrrVGsB+xWxhgtmCVptdqQACZjJUcr1A6M-cD4sEns9HklbPhKuq4AAxCD2AGw0qyxEQEAAXzLQA

File: tests/format/markdown/splitCjkText/format.test.js [hunk 2]
```diff
@@ -12,7 +12,8 @@ const code = Array.from({ length: 6 })
   .join("");
 // (39 + 40 + 41) * 2 === 40 * 3 * 2 === 40 * 6
 // lineWidth is 80 and "文" is double-width
-const output = `${"文".repeat(40)}\n`.repeat(6);
+// This is being saved for after the WebKit-based browser bug is fixed.
```
[the remark this answers is not in the document]
Comment (tats-u) [reply tests/format/markdown/splitCjkText/format.test.js]: makes sense.

**Review by sosukesuzuki — APPROVED** [review]
Thank you, LGTM

**Review by fisker — APPROVED** [review]

**Review by Thompson1985 — APPROVED** [review]

**Pull request merged**